library/core/src/iter/traits/iterator.rs

```diff
@@ -974,6 +974,7 @@ pub trait Iterator {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     #[rustc_do_not_const_check]
+    #[cfg_attr(not(test), rustc_diagnostic_item = "enumerate_method")]
```

Comment:
The _Rust Compiler Development Guide_ [explains](https://rustc-dev-guide.rust-lang.org/diagnostics/diagnostic-items.html#adding-diagnostic-items), not including `not(test)` can cause compilation errors while running some tests.  It goes on to say it's okay to add it as a preventative measure for all diagnostic items.

I did not test whether this specific diagnostic item causes any testing compilation errors.  I will look into this further to determine if it does.

Reply:
I ran `./x test`, which I believe runs all tests given my current configuration.  There were no failures with `not(test)` removed.

Reply:
The `cfg(not(test))` is required in `alloc` and `std`, but not in `core`.

This is because `alloc` and `std` contain some unit `#[test]`s as part of the library crates and the crates therefore have to be compiled twice: Once as a library as a dependency of the `test` crate (which contains the default test runner) and once as an executable that imports the `test` crate.

The `core` crate explicitly doesn't test it's library and only has integration tests, so the `cfg(not(test))` is not required there.
https://github.com/rust-lang/rust/blob/20aa2d81e36036073a9acf418c7d413cb4b22fa6/library/core/Cargo.toml#L14-L16

Reply:
Since `cfg(not(test))` is not required for this diagnostic item, should I remove it?  Or is it okay to leave it on as a preventative measure?